src/core/ext/filters/client_channel/lb_policy_factory.h

```diff
@@ -66,9 +66,18 @@ typedef struct grpc_lb_addresses {
 grpc_lb_addresses *grpc_lb_addresses_create(
     size_t num_addresses, const grpc_lb_user_data_vtable *user_data_vtable);
 
+/** Deep copy the address from \a src to \a dest. */
+void grpc_lb_address_copy(grpc_lb_address* dest, const grpc_lb_address* src,  
+                          const grpc_lb_user_data_vtable* user_data_vtable);
+
 /** Creates a copy of \a addresses. */
 grpc_lb_addresses *grpc_lb_addresses_copy(const grpc_lb_addresses *addresses);
 
+/** Extract the backend addresses and return them in a new \a 
+    grpc_lb_addresses. */
+grpc_lb_addresses *grpc_lb_addresses_extract_backends(
```

Comment:
Suggest calling this `grpc_lb_addresses_copy_backends()`.

Alternatively, instead of creating a new method for this, we could just add a parameter to `grpc_lb_addresses_copy()` that says whether to copy only backend addresses.